src/lasp_dependence_dag.erl

```diff
@@ -318,6 +320,89 @@ get_direct_edges(G, V1, V2) ->
         end
     end, digraph:out_edges(G, V1)).
 
+%% @doc Direct edges linking V1 and V2 that represent a process with an arity.
+-spec get_direct_edges_with_arity(digraph:graph(), id(), id(), pos_integer()) -> [edge()].
+get_direct_edges_with_arity(G, V1, V2, Arity) ->
+    lists:filter(fun({_, _, _, L}) ->
+        fun_arity(L#edge_label.transform) =:= Arity
+    end, get_direct_edges(G, V1, V2)).
+
+%% @doc Collect all transform functions and associated Pids from a list of edges.
+-spec collect_transform_functions([edge()]) -> {[function()], [pid()]}.
+collect_transform_functions(Edges) ->
+    lists:foldl(fun({_, _, _, Label}, {Fs, Pids}) ->
+        {[Label#edge_label.transform | Fs], [Label#edge_label.pid | Pids]}
+    end, {[], []}, Edges).
+
+%% @doc Merge all edges between Src and Dst into one single edge.
+%%
+%%      For each edge connecting Src and Dst that represents a
+%%      function of arity 1, collect all transforming functions
+%%      into a new process that represents the union of all of them.
+%%
+-spec merge_unary(id(), id(), digraph:graph()) -> ok | pid().
+merge_unary(Src, Dst, G) ->
+    Edges = get_direct_edges_with_arity(G, Src, Dst, 1),
+    union_unary(Edges).
+
+%% @doc Given a list of edges, merge them together into a single one.
+%%
+%%      Collect all transform functions into a new one that performs
+%%      the union of all results. Then kill all old Pids and start a
+%%      new lasp process with the mentioned function.
+%%
+%%      Note: assumes all read and write functions are the same in all
+%%      edges.
+%%
+%%      @todo add support for deltas (see map:core/6)
+%%
+-spec union_unary([edge()]) -> ok | pid().
+union_unary([{_, Src, Dst, L}, _ | _]=Edges) ->
+    %% If we have less than 2 edges, we are already optimized.
```

Comment:
`union_unary` is called with the edges that represent unary functions between two
vertices. As the goal of this is to reduce this number of edges to one, it only kicks
in when we have 2 or more edges.